Signal/src/call/OutboundCallInitiator.swift

```diff
@@ -46,7 +46,30 @@ import Foundation
             return false
         }
 
-        callUIAdapter.startAndShowOutgoingCall(recipientId: recipientId)
+        
+        // Check for microphone permissions
+        // Alternative way without prompting for permissions:
+        // if AVAudioSession.sharedInstance().recordPermission() == .denied {
+        AVAudioSession.sharedInstance().requestRecordPermission { isGranted in
+            // Here the permissions are either granted or denied
+            guard isGranted == true else {
+                Logger.warn("\(self.TAG) aborting due to missing microphone permissions.")
+                self.showAlertNoPermission()
+                return
+            }
+            callUIAdapter.startAndShowOutgoingCall(recipientId: recipientId)
+        }
         return true
     }
+    
+    /// Cleanup and present alert for no permissions
+    private func showAlertNoPermission() {
```

Comment:
nit: clarify that we're talking about microphone, not, e.g. video. 

maybe `func showNoMicrophonePermssionAlert`